cypress/support/step-definitions/decimal-steps.js

```diff
@@ -1,65 +1,70 @@
 import { inputPrecisionSlider } from '../../locators/decimal/index';
 import { setSlidebar } from '../helper';
-import { label, commonDataElementInputPreview } from '../../locators';
+import { label, commonDataElementInputPreviewNoIframe, getDataElementByValueNoIframe } from '../../locators';
 
 const TEXT_ALIGN = 'text-align';
 
 Then('input direction is {string}', (direction) => {
-  commonDataElementInputPreview().should('have.css', TEXT_ALIGN, `${direction}`);
+  commonDataElementInputPreviewNoIframe().should('have.css', TEXT_ALIGN, `${direction}`);
 });
 
 Then('Decimal component is disabled', () => {
-  commonDataElementInputPreview().should('have.attr', 'disabled');
-  commonDataElementInputPreview().parent().should('have.attr', 'disabled');
+  commonDataElementInputPreviewNoIframe().should('have.attr', 'disabled');
+  commonDataElementInputPreviewNoIframe().parent().should('have.attr', 'disabled');
 });
 
 Then('Decimal component is enabled', () => {
-  commonDataElementInputPreview().should('not.have.attr', 'disabled');
-  commonDataElementInputPreview().parent().should('not.have.attr', 'disabled');
+  commonDataElementInputPreviewNoIframe().should('not.have.attr', 'disabled');
+  commonDataElementInputPreviewNoIframe().parent().should('not.have.attr', 'disabled');
 });
 
 Then('Decimal component is readOnly', () => {
-  commonDataElementInputPreview().should('have.attr', 'readonly');
-  commonDataElementInputPreview().parent().should('have.attr', 'readonly');
+  commonDataElementInputPreviewNoIframe().should('have.attr', 'readonly');
+  commonDataElementInputPreviewNoIframe().parent().should('have.attr', 'readonly');
 });
 
 Then('Decimal component is not readOnly', () => {
-  commonDataElementInputPreview().should('not.have.attr', 'readonly');
-  commonDataElementInputPreview().parent().should('not.have.attr', 'readonly');
+  commonDataElementInputPreviewNoIframe().should('not.have.attr', 'readonly');
+  commonDataElementInputPreviewNoIframe().parent().should('not.have.attr', 'readonly');
 });
 
-When('label width on preview is {int}', (width) => {
+Then('label width on preview is {int}', (width) => {
+  getDataElementByValueNoIframe('label').should('have.attr', 'width').should('contain', `${width}`);
+});
+
+Then('label width on preview is {int} in IFrame', (width) => {
   label().should('have.attr', 'width').should('contain', `${width}`);
 });
 
-When('inputWidth on preview is {int}', (width) => {
-  commonDataElementInputPreview().parent().should('have.css', 'flex').should('contain', `${width}%`);
+Then('inputWidth on preview is {int}', (width) => {
```

Comment:
IF it's only for `Decimal`, please rename this step e.g `Decimal input direction is {string}`. If it's a reusable step for other components, move it to common file

Reply:
moved to `common-steps.js` file